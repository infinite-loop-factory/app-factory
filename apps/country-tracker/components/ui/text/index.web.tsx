import type { VariantProps } from "@gluestack-ui/nativewind-utils";
import type { ComponentProps, ElementRef } from "react";

import { forwardRef } from "react";
import { textStyle } from "./styles";

type ITextProps = ComponentProps<"span"> & VariantProps<typeof textStyle>;

const Text = forwardRef<ElementRef<"span">, ITextProps>(
  (
    {
      className,
      isTruncated,
      bold,
      underline,
      strikeThrough,
      size = "md",
      sub,
      italic,
      highlight,
      ...props
    }: { className?: string } & ITextProps,
    ref,
  ) => {
    return (
      <span
        className={textStyle({
          isTruncated,
          bold,
          underline,
          strikeThrough,
          size,
          sub,
          italic,
          highlight,
          class: className,
        })}
        {...props}
        ref={ref}
      />
    );
  },
);

Text.displayName = "Text";

export { Text };
