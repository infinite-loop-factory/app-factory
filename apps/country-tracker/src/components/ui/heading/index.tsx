import type { VariantProps } from "@gluestack-ui/utils/nativewind-utils";

import { H1, H2, H3, H4, H5, H6 } from "@expo/html-elements";
import { cssInterop } from "nativewind";
import React, { forwardRef, memo } from "react";
import { createVariantResolver } from "@/utils/variant-resolver";
import { headingStyle } from "./styles";

const resolveHeadingSize = createVariantResolver([
  "5xl",
  "4xl",
  "3xl",
  "2xl",
  "xl",
  "lg",
  "md",
  "sm",
  "xs",
] as const);

type IHeadingProps = VariantProps<typeof headingStyle> &
  React.ComponentPropsWithoutRef<typeof H1> & {
    as?: React.ElementType;
  };

cssInterop(H1, { className: "style" });
cssInterop(H2, { className: "style" });
cssInterop(H3, { className: "style" });
cssInterop(H4, { className: "style" });
cssInterop(H5, { className: "style" });
cssInterop(H6, { className: "style" });

const MappedHeading = memo(
  forwardRef<React.ComponentRef<typeof H1>, IHeadingProps>(
    function MappedHeading(
      {
        size,
        className,
        isTruncated,
        bold,
        underline,
        strikeThrough,
        sub,
        italic,
        highlight,
        ...props
      },
      ref,
    ) {
      const variantSize = resolveHeadingSize(size);
      const effectiveSize = variantSize ?? "lg";
      const computedClass = headingStyle({
        size: variantSize,
        isTruncated: Boolean(isTruncated),
        bold: Boolean(bold),
        underline: Boolean(underline),
        strikeThrough: Boolean(strikeThrough),
        sub: Boolean(sub),
        italic: Boolean(italic),
        highlight: Boolean(highlight),
        class: className,
      });

      switch (effectiveSize) {
        case "5xl":
        case "4xl":
        case "3xl":
          return (
            <H1
              className={computedClass}
              {...props}
              // @ts-expect-error : type issue
              ref={ref}
            />
          );
        case "2xl":
          return (
            <H2
              className={computedClass}
              {...props}
              // @ts-expect-error : type issue
              ref={ref}
            />
          );
        case "xl":
          return (
            <H3
              className={computedClass}
              {...props}
              // @ts-expect-error : type issue
              ref={ref}
            />
          );
        case "lg":
          return (
            <H4
              className={computedClass}
              {...props}
              // @ts-expect-error : type issue
              ref={ref}
            />
          );
        case "md":
          return (
            <H5
              className={computedClass}
              {...props}
              // @ts-expect-error : type issue
              ref={ref}
            />
          );
        case "sm":
        case "xs":
          return (
            <H6
              className={computedClass}
              {...props}
              // @ts-expect-error : type issue
              ref={ref}
            />
          );
        default:
          return (
            <H4
              className={computedClass}
              {...props}
              // @ts-expect-error : type issue
              ref={ref}
            />
          );
      }
    },
  ),
);

const Heading = memo(
  forwardRef<React.ComponentRef<typeof H1>, IHeadingProps>(function Heading(
    { className, size = "lg", as: AsComp, ...props },
    ref,
  ) {
    const variantSize = resolveHeadingSize(size) ?? "lg";
    const {
      isTruncated,
      bold,
      underline,
      strikeThrough,
      sub,
      italic,
      highlight,
    } = props;

    if (AsComp) {
      return (
        <AsComp
          className={headingStyle({
            size: variantSize,
            isTruncated: Boolean(isTruncated),
            bold: Boolean(bold),
            underline: Boolean(underline),
            strikeThrough: Boolean(strikeThrough),
            sub: Boolean(sub),
            italic: Boolean(italic),
            highlight: Boolean(highlight),
            class: className,
          })}
          {...props}
        />
      );
    }

    return (
      <MappedHeading
        className={className}
        ref={ref}
        size={variantSize}
        {...props}
      />
    );
  }),
);

Heading.displayName = "Heading";

export { Heading };
