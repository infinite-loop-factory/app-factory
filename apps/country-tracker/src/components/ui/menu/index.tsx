"use client";
import type { VariantProps } from "@gluestack-ui/utils/nativewind-utils";

import { createMenu } from "@gluestack-ui/core/menu/creator";
import { tva } from "@gluestack-ui/utils/nativewind-utils";
import {
  AnimatePresence,
  Motion,
  MotionComponentProps,
} from "@legendapp/motion";
import { cssInterop } from "nativewind";
import React from "react";
import { Pressable, Text, View, ViewStyle } from "react-native";
import { createVariantResolver } from "@/utils/variant-resolver";

type IMotionViewProps = React.ComponentProps<typeof View> &
  MotionComponentProps<typeof View, ViewStyle, unknown, unknown, unknown>;

const MotionView = Motion.View as React.ComponentType<IMotionViewProps>;

const menuStyle = tva({
  base: "rounded-md bg-background-0 border border-outline-100 p-1 shadow-hard-5",
});

const menuItemStyle = tva({
  base: "min-w-[200px] p-3 flex-row items-center rounded data-[hover=true]:bg-background-50 data-[active=true]:bg-background-100 data-[focus=true]:bg-background-50 data-[focus=true]:web:outline-none data-[focus=true]:web:outline-0 data-[disabled=true]:opacity-40 data-[disabled=true]:web:cursor-not-allowed data-[focus-visible=true]:web:outline-2 data-[focus-visible=true]:web:outline-primary-700 data-[focus-visible=true]:web:outline data-[focus-visible=true]:web:cursor-pointer data-[disabled=true]:data-[focus=true]:bg-transparent",
});

const menuBackdropStyle = tva({
  base: "absolute top-0 bottom-0 left-0 right-0 web:cursor-default",
  // add this classnames if you want to give background color to backdrop
  // opacity-50 bg-background-500,
});

const menuSeparatorStyle = tva({
  base: "bg-background-200 h-px w-full",
});

const menuItemLabelStyle = tva({
  base: "text-typography-700 font-normal font-body",

  variants: {
    isTruncated: {
      true: "web:truncate",
    },
    bold: {
      true: "font-bold",
    },
    underline: {
      true: "underline",
    },
    strikeThrough: {
      true: "line-through",
    },
    size: {
      "2xs": "text-2xs",
      xs: "text-xs",
      sm: "text-sm",
      md: "text-base",
      lg: "text-lg",
      xl: "text-xl",
      "2xl": "text-2xl",
      "3xl": "text-3xl",
      "4xl": "text-4xl",
      "5xl": "text-5xl",
      "6xl": "text-6xl",
    },
    sub: {
      true: "text-xs",
    },
    italic: {
      true: "italic",
    },
    highlight: {
      true: "bg-yellow-500",
    },
  },
});

const resolveMenuItemLabelSize = createVariantResolver([
  "2xs",
  "xs",
  "sm",
  "md",
  "lg",
  "xl",
  "2xl",
  "3xl",
  "4xl",
  "5xl",
  "6xl",
] as const);

const BackdropPressable = React.forwardRef<
  React.ComponentRef<typeof Pressable>,
  React.ComponentPropsWithoutRef<typeof Pressable> &
    VariantProps<typeof menuBackdropStyle>
>(function BackdropPressable({ className, ...props }, ref) {
  return (
    <Pressable
      className={menuBackdropStyle({
        class: className,
      })}
      ref={ref}
      {...props}
    />
  );
});

type IMenuItemProps = VariantProps<typeof menuItemStyle> & {
  className?: string;
} & React.ComponentPropsWithoutRef<typeof Pressable>;

const Item = React.forwardRef<
  React.ComponentRef<typeof Pressable>,
  IMenuItemProps
>(function Item({ className, ...props }, ref) {
  return (
    <Pressable
      className={menuItemStyle({
        class: className,
      })}
      ref={ref}
      {...props}
    />
  );
});

const Separator = React.forwardRef<
  React.ComponentRef<typeof View>,
  React.ComponentPropsWithoutRef<typeof View> &
    VariantProps<typeof menuSeparatorStyle>
>(function Separator({ className, ...props }, ref) {
  return (
    <View
      className={menuSeparatorStyle({ class: className })}
      ref={ref}
      {...props}
    />
  );
});
export const UIMenu = createMenu({
  Root: MotionView,
  Item: Item,
  Label: Text,
  Backdrop: BackdropPressable,
  AnimatePresence: AnimatePresence,
  Separator: Separator,
});

cssInterop(MotionView, { className: "style" });

type IMenuProps = React.ComponentProps<typeof UIMenu> &
  VariantProps<typeof menuStyle> & { className?: string };
type IMenuItemLabelProps = React.ComponentProps<typeof UIMenu.ItemLabel> &
  VariantProps<typeof menuItemLabelStyle> & { className?: string };

const Menu = React.forwardRef<React.ComponentRef<typeof UIMenu>, IMenuProps>(
  function Menu({ className, ...props }, ref) {
    return (
      <UIMenu
        animate={{
          opacity: 1,
          scale: 1,
        }}
        className={menuStyle({
          class: className,
        })}
        exit={{
          opacity: 0,
          scale: 0.8,
        }}
        initial={{
          opacity: 0,
          scale: 0.8,
        }}
        ref={ref}
        transition={{
          type: "timing",
          duration: 100,
        }}
        {...props}
      />
    );
  },
);

const MenuItem = UIMenu.Item;

const MenuItemLabel = React.forwardRef<
  React.ComponentRef<typeof UIMenu.ItemLabel>,
  IMenuItemLabelProps
>(function MenuItemLabel(
  {
    className,
    isTruncated,
    bold,
    underline,
    strikeThrough,
    size = "md",
    sub,
    italic,
    highlight,
    ...props
  },
  ref,
) {
  const variantSize = resolveMenuItemLabelSize(size);
  return (
    <UIMenu.ItemLabel
      className={menuItemLabelStyle({
        isTruncated: Boolean(isTruncated),
        bold: Boolean(bold),
        underline: Boolean(underline),
        strikeThrough: Boolean(strikeThrough),
        size: variantSize,
        sub: Boolean(sub),
        italic: Boolean(italic),
        highlight: Boolean(highlight),
        class: className,
      })}
      ref={ref}
      {...props}
    />
  );
});

const MenuSeparator = UIMenu.Separator;

Menu.displayName = "Menu";
MenuItem.displayName = "MenuItem";
MenuItemLabel.displayName = "MenuItemLabel";
MenuSeparator.displayName = "MenuSeparator";
export { Menu, MenuItem, MenuItemLabel, MenuSeparator };
