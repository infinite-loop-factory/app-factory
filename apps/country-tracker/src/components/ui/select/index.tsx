"use client";

import type { VariantProps } from "@gluestack-ui/utils/nativewind-utils";

import { PrimitiveIcon, UIIcon } from "@gluestack-ui/core/icon/creator";
import { createSelect } from "@gluestack-ui/core/select/creator";
import {
  tva,
  useStyleContext,
  withStyleContext,
} from "@gluestack-ui/utils/nativewind-utils";
import { cssInterop } from "nativewind";
import React from "react";
import { Pressable, TextInput, View } from "react-native";
import { createVariantResolver } from "@/utils/variant-resolver";
import {
  Actionsheet,
  ActionsheetBackdrop,
  ActionsheetContent,
  ActionsheetDragIndicator,
  ActionsheetDragIndicatorWrapper,
  ActionsheetFlatList,
  ActionsheetItem,
  ActionsheetItemText,
  ActionsheetScrollView,
  ActionsheetSectionHeaderText,
  ActionsheetSectionList,
  ActionsheetVirtualizedList,
} from "./select-actionsheet";

const SelectTriggerWrapper = React.forwardRef<
  React.ComponentRef<typeof Pressable>,
  React.ComponentProps<typeof Pressable>
>(function SelectTriggerWrapper({ ...props }, ref) {
  return <Pressable {...props} ref={ref} />;
});

const selectIconStyle = tva({
  base: "text-background-500 fill-none",
  parentVariants: {
    size: {
      "2xs": "h-3 w-3",
      xs: "h-3.5 w-3.5",
      sm: "h-4 w-4",
      md: "h-[18px] w-[18px]",
      lg: "h-5 w-5",
      xl: "h-6 w-6",
    },
  },
});

const resolveSelectIconSize = createVariantResolver([
  "2xs",
  "xs",
  "sm",
  "md",
  "lg",
  "xl",
] as const);

const selectStyle = tva({
  base: "",
});

const selectTriggerStyle = tva({
  base: "border border-background-300 rounded flex-row items-center overflow-hidden data-[hover=true]:border-outline-400 data-[focus=true]:border-primary-700 data-[disabled=true]:opacity-40 data-[disabled=true]:data-[hover=true]:border-background-300",
  variants: {
    size: {
      xl: "h-12",
      lg: "h-11",
      md: "h-10",
      sm: "h-9",
    },
    variant: {
      underlined:
        "border-0 border-b rounded-none data-[hover=true]:border-primary-700 data-[focus=true]:border-primary-700 data-[focus=true]:web:shadow-[inset_0_-1px_0_0] data-[focus=true]:web:shadow-primary-700 data-[invalid=true]:border-error-700 data-[invalid=true]:web:shadow-error-700",
      outline:
        "data-[focus=true]:border-primary-700 data-[focus=true]:web:shadow-[inset_0_0_0_1px] data-[focus=true]:data-[hover=true]:web:shadow-primary-600 data-[invalid=true]:web:shadow-[inset_0_0_0_1px] data-[invalid=true]:border-error-700 data-[invalid=true]:web:shadow-error-700 data-[invalid=true]:data-[hover=true]:border-error-700",
      rounded:
        "rounded-full data-[focus=true]:border-primary-700 data-[focus=true]:web:shadow-[inset_0_0_0_1px] data-[focus=true]:web:shadow-primary-700 data-[invalid=true]:border-error-700 data-[invalid=true]:web:shadow-error-700",
    },
  },
});

const resolveSelectTriggerSize = createVariantResolver([
  "xl",
  "lg",
  "md",
  "sm",
] as const);

const resolveSelectTriggerVariant = createVariantResolver([
  "underlined",
  "outline",
  "rounded",
] as const);

const selectInputStyle = tva({
  base: "px-3 placeholder:text-typography-500 web:w-full h-full text-typography-900 pointer-events-none web:outline-none ios:leading-[0px] py-0",
  parentVariants: {
    size: {
      xl: "text-xl",
      lg: "text-lg",
      md: "text-base",
      sm: "text-sm",
    },
    variant: {
      underlined: "px-0",
      outline: "",
      rounded: "px-4",
    },
  },
});

const resolveSelectInputSize = createVariantResolver([
  "xl",
  "lg",
  "md",
  "sm",
] as const);

const resolveSelectInputVariant = createVariantResolver([
  "underlined",
  "outline",
  "rounded",
] as const);

const UISelect = createSelect(
  {
    Root: View,
    Trigger: withStyleContext(SelectTriggerWrapper),
    Input: TextInput,
    Icon: UIIcon,
  },
  {
    Portal: Actionsheet,
    Backdrop: ActionsheetBackdrop,
    Content: ActionsheetContent,
    DragIndicator: ActionsheetDragIndicator,
    DragIndicatorWrapper: ActionsheetDragIndicatorWrapper,
    Item: ActionsheetItem,
    ItemText: ActionsheetItemText,
    ScrollView: ActionsheetScrollView,
    VirtualizedList: ActionsheetVirtualizedList,
    FlatList: ActionsheetFlatList,
    SectionList: ActionsheetSectionList,
    SectionHeaderText: ActionsheetSectionHeaderText,
  },
);

cssInterop(UISelect, { className: "style" });
cssInterop(UISelect.Input, {
  className: { target: "style", nativeStyleToProp: { textAlign: true } },
});
cssInterop(SelectTriggerWrapper, { className: "style" });

cssInterop(PrimitiveIcon, {
  className: {
    target: "style",
    nativeStyleToProp: {
      height: true,
      width: true,
      fill: true,
      color: "classNameColor",
      stroke: true,
    },
  },
});

type ISelectProps = VariantProps<typeof selectStyle> &
  React.ComponentProps<typeof UISelect> & { className?: string };

const Select = React.forwardRef<
  React.ComponentRef<typeof UISelect>,
  ISelectProps
>(function Select({ className, ...props }, ref) {
  return (
    <UISelect
      className={selectStyle({
        class: className,
      })}
      ref={ref}
      {...props}
    />
  );
});

type ISelectTriggerProps = VariantProps<typeof selectTriggerStyle> &
  React.ComponentProps<typeof UISelect.Trigger> & { className?: string };

const SelectTrigger = React.forwardRef<
  React.ComponentRef<typeof UISelect.Trigger>,
  ISelectTriggerProps
>(function SelectTrigger(
  { className, size = "md", variant = "outline", ...props },
  ref,
) {
  const variantSize = resolveSelectTriggerSize(size);
  const variantStyle = resolveSelectTriggerVariant(variant);
  return (
    <UISelect.Trigger
      className={selectTriggerStyle({
        class: className,
        size: variantSize,
        variant: variantStyle,
      })}
      context={{
        size: variantSize ?? size,
        variant: variantStyle ?? variant,
      }}
      ref={ref}
      {...props}
    />
  );
});

type ISelectInputProps = VariantProps<typeof selectInputStyle> &
  React.ComponentProps<typeof UISelect.Input> & { className?: string };

const SelectInput = React.forwardRef<
  React.ComponentRef<typeof UISelect.Input>,
  ISelectInputProps
>(function SelectInput({ className, ...props }, ref) {
  const { size: parentSize, variant: parentVariant } = useStyleContext();
  const resolvedParentSize = resolveSelectInputSize(parentSize);
  const resolvedParentVariant = resolveSelectInputVariant(parentVariant);
  return (
    <UISelect.Input
      className={selectInputStyle({
        class: className,
        parentVariants: {
          size: resolvedParentSize,
          variant: resolvedParentVariant,
        },
      })}
      ref={ref}
      {...props}
    />
  );
});

type ISelectIcon = VariantProps<typeof selectIconStyle> &
  React.ComponentProps<typeof UISelect.Icon> & {
    className?: string;
    height?: number;
    width?: number;
  };

const SelectIcon = React.forwardRef<
  React.ComponentRef<typeof UISelect.Icon>,
  ISelectIcon
>(function SelectIcon({ className, size, height, width, ...props }, ref) {
  const { size: parentSize } = useStyleContext();
  if (typeof size === "number") {
    return (
      <UISelect.Icon
        ref={ref}
        {...props}
        className={selectIconStyle({ class: className })}
        height={height}
        size={size}
        width={width}
      />
    );
  } else if (
    (height !== undefined || width !== undefined) &&
    size === undefined
  ) {
    return (
      <UISelect.Icon
        ref={ref}
        {...props}
        className={selectIconStyle({ class: className })}
        height={height}
        width={width}
      />
    );
  }
  const variantSize = resolveSelectIconSize(size);
  const parentVariantSize = resolveSelectIconSize(parentSize);
  return (
    <UISelect.Icon
      className={selectIconStyle({
        class: className,
        size: variantSize,
        parentVariants: {
          size: parentVariantSize,
        },
      })}
      height={height}
      ref={ref}
      size={size}
      width={width}
      {...props}
    />
  );
});

Select.displayName = "Select";
SelectTrigger.displayName = "SelectTrigger";
SelectInput.displayName = "SelectInput";
SelectIcon.displayName = "SelectIcon";

// Actionsheet Components
const SelectPortal = UISelect.Portal;
const SelectBackdrop = UISelect.Backdrop;
const SelectContent = UISelect.Content;
const SelectDragIndicator = UISelect.DragIndicator;
const SelectDragIndicatorWrapper = UISelect.DragIndicatorWrapper;
const SelectItem = UISelect.Item;
const SelectScrollView = UISelect.ScrollView;
const SelectVirtualizedList = UISelect.VirtualizedList;
const SelectFlatList = UISelect.FlatList;
const SelectSectionList = UISelect.SectionList;
const SelectSectionHeaderText = UISelect.SectionHeaderText;

export {
  Select,
  SelectTrigger,
  SelectInput,
  SelectIcon,
  SelectPortal,
  SelectBackdrop,
  SelectContent,
  SelectDragIndicator,
  SelectDragIndicatorWrapper,
  SelectItem,
  SelectScrollView,
  SelectVirtualizedList,
  SelectFlatList,
  SelectSectionList,
  SelectSectionHeaderText,
};
