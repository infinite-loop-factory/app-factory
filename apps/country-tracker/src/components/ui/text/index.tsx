import type { VariantProps } from "@gluestack-ui/utils/nativewind-utils";

import React from "react";
import { Text as RNText } from "react-native";
import { createVariantResolver } from "@/utils/variant-resolver";
import { textStyle } from "./styles";

type ITextProps = React.ComponentProps<typeof RNText> &
  VariantProps<typeof textStyle>;

const resolveTextSize = createVariantResolver([
  "2xs",
  "xs",
  "sm",
  "md",
  "lg",
  "xl",
  "2xl",
  "3xl",
  "4xl",
  "5xl",
  "6xl",
] as const);

const Text = React.forwardRef<React.ComponentRef<typeof RNText>, ITextProps>(
  function Text(
    {
      className,
      isTruncated,
      bold,
      underline,
      strikeThrough,
      size = "md",
      sub,
      italic,
      highlight,
      ...props
    },
    ref,
  ) {
    const variantSize = resolveTextSize(size);
    return (
      <RNText
        className={textStyle({
          isTruncated: Boolean(isTruncated),
          bold: Boolean(bold),
          underline: Boolean(underline),
          strikeThrough: Boolean(strikeThrough),
          size: variantSize,
          sub: Boolean(sub),
          italic: Boolean(italic),
          highlight: Boolean(highlight),
          class: className,
        })}
        {...props}
        ref={ref}
      />
    );
  },
);

Text.displayName = "Text";

export { Text };
