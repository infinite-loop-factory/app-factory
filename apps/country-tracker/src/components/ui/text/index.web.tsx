import type { VariantProps } from "@gluestack-ui/utils/nativewind-utils";

import React from "react";
import { createVariantResolver } from "@/utils/variant-resolver";
import { textStyle } from "./styles";

type ITextProps = React.ComponentProps<"span"> & VariantProps<typeof textStyle>;

const resolveTextSize = createVariantResolver([
  "2xs",
  "xs",
  "sm",
  "md",
  "lg",
  "xl",
  "2xl",
  "3xl",
  "4xl",
  "5xl",
  "6xl",
] as const);

const Text = React.forwardRef<React.ComponentRef<"span">, ITextProps>(
  function Text(
    {
      className,
      isTruncated,
      bold,
      underline,
      strikeThrough,
      size = "md",
      sub,
      italic,
      highlight,
      ...props
    }: { className?: string } & ITextProps,
    ref,
  ) {
    const variantSize = resolveTextSize(size);
    return (
      <span
        className={textStyle({
          isTruncated: Boolean(isTruncated),
          bold: Boolean(bold),
          underline: Boolean(underline),
          strikeThrough: Boolean(strikeThrough),
          size: variantSize,
          sub: Boolean(sub),
          italic: Boolean(italic),
          highlight: Boolean(highlight),
          class: className,
        })}
        {...props}
        ref={ref}
      />
    );
  },
);

Text.displayName = "Text";

export { Text };
