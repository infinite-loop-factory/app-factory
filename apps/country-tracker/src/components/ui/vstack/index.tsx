import type { VariantProps } from "@gluestack-ui/utils/nativewind-utils";

import React from "react";
import { View } from "react-native";
import { createVariantResolver } from "@/utils/variant-resolver";
import { vstackStyle } from "./styles";

const resolveVstackSpace = createVariantResolver([
  "xs",
  "sm",
  "md",
  "lg",
  "xl",
  "2xl",
  "3xl",
  "4xl",
] as const);

type IVStackProps = React.ComponentProps<typeof View> &
  VariantProps<typeof vstackStyle>;

const VStack = React.forwardRef<React.ComponentRef<typeof View>, IVStackProps>(
  function VStack({ className, space, reversed, ...props }, ref) {
    return (
      <View
        className={vstackStyle({
          space: resolveVstackSpace(space),
          reversed: reversed as boolean,
          class: className,
        })}
        {...props}
        ref={ref}
      />
    );
  },
);

VStack.displayName = "VStack";

export { VStack };
