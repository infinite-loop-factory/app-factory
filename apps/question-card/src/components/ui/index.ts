/**
 * Gluestack UI Components Index
 * 모든 UI 컴포넌트를 중앙에서 export
 */

// 플로팅 UI 타입 exports
export type {
  FloatingActionButtonPosition,
  FloatingActionButtonProps,
  FloatingActionButtonStyle,
} from "./FloatingActionButton";
export type {
  FloatingBackButtonPosition,
  FloatingBackButtonProps,
  FloatingBackButtonStyle,
} from "./FloatingBackButton";
export type {
  FloatingMenuButtonPosition,
  FloatingMenuButtonProps,
  FloatingMenuButtonStyle,
} from "./FloatingMenuButton";
export type { OrangeHeaderProps } from "./OrangeHeader";
export type { ProgressIndicatorProps } from "./ProgressIndicator";

export { Badge } from "./badge";
export { Box } from "./box";
export { Button } from "./button";
export { Card } from "./card";
export { Center } from "./center";
// 플로팅 UI 컴포넌트
export { FloatingActionButton } from "./FloatingActionButton";
export { FloatingBackButton } from "./FloatingBackButton";
export { FloatingMenuButton } from "./FloatingMenuButton";
export { GluestackUIProvider } from "./gluestack-ui-provider";
export { HStack } from "./hstack";
export { OrangeHeader } from "./OrangeHeader";
export { ProgressIndicator } from "./ProgressIndicator";
export { Pressable } from "./pressable";
export { Progress } from "./progress";
export { Text } from "./text";
export { VStack } from "./vstack";
