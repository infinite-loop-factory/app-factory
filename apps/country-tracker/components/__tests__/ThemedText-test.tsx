import renderer from "react-test-renderer";

import { Text } from "@/components/ui/text";

it("renders correctly", () => {
  const tree = renderer.create(<Text>Snapshot test!</Text>).toJSON();

  expect(tree).toMatchSnapshot();
});
