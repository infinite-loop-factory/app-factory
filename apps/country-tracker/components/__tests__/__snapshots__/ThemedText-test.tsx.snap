// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  className="text-typography-700 font-body text-base"
>
  Snapshot test!
</Text>
`;
